AchieveIt.API/Extensions/ExceptionMiddlewareExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Net;
+using AchieveIt.Shared.Exceptions;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+
+namespace AchieveIt.API.Extensions
+{
+    public static class ExceptionMiddlewareExtensions
+    {
+        public static void ConfigureExceptionHandler(this IApplicationBuilder applicationBuilder)
+        {
+            applicationBuilder.UseExceptionHandler(appException =>
+            {
+                appException.Run(async context =>
+                {
+                    context.Response.StatusCode = (int) HttpStatusCode.InternalServerError;
+                    context.Response.ContentType = "application/json";
+
+                    var contextFeature = context.Features.Get<IExceptionHandlerFeature>();
+                    if (contextFeature != null)
+                    {
+                        await context.Response.WriteAsync(new ExceptionDetails
+                        {
+                            StatusCode = context.Response.StatusCode,
+                            Message = "Internal Server Error"
```

Comment:
It should contain the error details or message...

Reply:
Done.